modules/cli/src/main/scala/scala/cli/commands/Package.scala

```diff
@@ -58,8 +59,16 @@ object Package extends ScalaCommand[PackageOptions] {
       ) { res =>
         res.orReport(logger).map(_.main).foreach {
           case s: Build.Successful =>
-            doPackage(inputs, logger, options.output.filter(_.nonEmpty), options.force, s)
+            val mtimeDestPath = doPackage(
+              inputs,
+              logger,
+              options.output.filter(_.nonEmpty),
+              options.force,
+              s,
+              expectedModifyEpochSecondOpt
+            )
               .orReport(logger)
+            expectedModifyEpochSecondOpt = mtimeDestPath.flatten
```

Comment:
Let's not make `expectedModifyEpochSecondOpt` empty if the last run just failed (because compilation failed, or anything else). Something like this should retain the value of the last successful run:
```suggestion
            for (valueOpt <- mtimeDestPath)
              expectedModifyEpochSecondOpt = valueOpt
```